omnisend/includes/SDK/V1/class-contact.php

```diff
@@ -224,6 +225,134 @@ public function to_array(): array {
 		return $arr;
 	}
 
+	/**
+	 * Convert contact to array.
+	 *
+	 * If contact is valid it will be transformed to array that can be sent to Omnisend.
+	 *
+	 * @return array
+	 */
+	public function to_array_for_save_contract(): array {
+		if ( $this->validate()->has_errors() ) {
+			return array();
+		}
+
+		$time_now = gmdate( 'c' );
+
+		$user_agent = sanitize_text_field( wp_unslash( $_SERVER['HTTP_USER_AGENT'] ?? 'user agent not found' ) );
+		$ip         = sanitize_text_field( wp_unslash( $_SERVER['REMOTE_ADDR'] ?? 'ip not found' ) );
+
+		$arr = array(
+			'identifiers' => array(),
+			'tags'        => array_values( array_unique( $this->tags ) ),
+		);
+
+		$email_consent_status = 'subscribed';
+		if ( $this->email_opt_out_source ) {
+			$email_consent_status = 'unsubscribed';
+		}
+
+		$sms_consent_status = 'subscribed';
+		if ( $this->email_opt_out_source ) {
```

Comment:
you are checking email settings here and editing sms consent

Reply:
Pakoregavau